pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -2226,6 +2226,53 @@ func (a *ApplicationBackupController) cleanupResources(
 		log.ApplicationBackupLog(backup).Errorf("%v", errMsg)
 		return err
 	}
+
```

Comment:
Please add comment here why we are doing here.
Also please comment why the previous ruleDelete would not have happened for VirtualMachine backupTypes.